revise HPA major theme (#1767)
kubernetes/sig-release: releases/release-1.23/release-notes/release-notes-draft.md

Although clusters are enabled to support dual-stack networking, Pods and Services continue to default to single-stack. 
To use dual-stack networking: Kubernetes nodes have routable IPv4/IPv6 network interfaces, a dual-stack capable CNI network plugin is used, Pods are configured to be dual-stack and Services have their `.spec.ipFamilyPolicy` field set to either `PreferDualStack` or `RequireDualStack`.

> ### The HorizontalPodAutoscaler v2 API is GA, the HorizontalPodAutoscaler v2beta2 API is deprecated

Reviewer: ```suggestion
### HorizontalPodAutoscaler v2 graduates to GA
```
The very next line mentions the deprecation; I think that's OK.

> The HorizontalPodAutscaler `autoscaling/v2` stable API is GA. The HorizontalPodAutoscaler `autoscaling/v2beta2` API is deprecated.

Reviewer: Fixing a typo plus rewording:
```suggestion
Version 2 of the HorizontalPodAutoscalerAPI graduates to stable in the 1.23 release. The HorizontalPodAutoscaler `autoscaling/v2beta2` API is deprecated in favor of the new `autoscaling/v2` API, which the Kubernetes project recommends for all use cases.

This release does *not* deprecate the v1 HorizontalPodAutoscaler API.
```
? (last line might or might not be correct)

Although clusters are enabled to support dual-stack networking, Pods and Services continue to default to single-stack. 
To use dual-stack networking: Kubernetes nodes have routable IPv4/IPv6 network interfaces, a dual-stack capable CNI network plugin is used, Pods are configured to be dual-stack and Services have their `.spec.ipFamilyPolicy` field set to either `PreferDualStack` or `RequireDualStack`.

### HorizontalPodAutoscaler v2 graduates to GA

> Version 2 of the HorizontalPodAutoscalerAPI graduates to stable in the 1.23 release. The HorizontalPodAutoscaler `autoscaling/v2beta2` API is deprecated in favor of the new `autoscaling/v2` API, which the Kubernetes project recommends for all use cases.

Reviewer: Late nit (sorry):
```suggestion
Version 2 of the HorizontalPodAutoscaler API graduates to stable in the 1.23 release. The HorizontalPodAutoscaler `autoscaling/v2beta2` API is deprecated in favor of the new `autoscaling/v2` API, which the Kubernetes project recommends for all use cases.
```

Author: Made follow-up PR https://github.com/kubernetes/sig-release/pull/1770
